output_aggregator.go

```diff
@@ -125,6 +127,44 @@ func (a *aggregator) Read(read []byte) (int, error) {
 	return buffer.Len() + 1, err
 }
 
+// WriteTo implements io.WriterTo, and returns int64 instead of int because of:
+// https://stackoverflow.com/questions/29658892/why-does-io-writertos-writeto-method-return-an-int64-rather-than-an-int
+func (a *aggregator) WriteTo(dst io.Writer) (int64, error) {
+	if len(a.filterFuncs) == 0 {
+		// Happy path, directly pipe output
+		doneC := make(chan int64)
+		go func() {
+			written, _ := io.Copy(dst, a.reader)
+			doneC <- written
+		}()
+		errC := make(chan error)
+		go func() {
+			errC <- a.Wait()
+		}()
+		return <-doneC, <-errC
+	}
+
+	// Pipe output via the filtered line pipe
+	filterErrC := make(chan error, 1)
+	writtenC := make(chan int64, 1)
```

Comment:
Unless I'm mistaken, we don't need the buffered channels here, if we read the error before reading the written count.

To put it differently, the need for the buffering comes from the fact that we're writing into A then B, while we're reading from B then A. Writing in A then B and reading in the same order removes the need for the buffered channel. 

All examples work with unbuffered channels when lines 152 and 153 are swapped.

Reply:
Done in the very messy 219a611d231a3988ea00a93bdc50c73c99231c64 !